src/Runtime/Core/Serialization/SerializationAttributes.cs

```diff
@@ -190,11 +210,10 @@ public static List<ConstructionAttribute> GetConstructionAttribute(Type type)
         // Order so that the UseWhen==null case comes last.
         public int CompareTo(object obj)
         {
-            ConstructionAttribute ca = obj as ConstructionAttribute;
-            if (ca == null) return -1;
-            if ((UseWhen == null) && (ca.UseWhen != null)) return 1;
-            if ((UseWhen != null) && (ca.UseWhen == null)) return -1;
-            return 0;
+            return obj is ConstructionAttribute that
+                ? that.Priority - this.Priority
```

Comment:
Just delegate using `this.Priority.CompareTo(that.Priority)`, we shouldn't use magic `-` operations, usually.

Reply:
Sure. Though the minus operator is the way these comparisons have been implemented in inferior languages (like C) for ages :) Actually that's the whole reason why CompareTo() returns int instead of enum.